frontend/src/util/toolbar_functions.js

```diff
@@ -1,5 +1,42 @@
-export function print() {
+import jsPDF from 'jspdf';
+import { getRedactionsOnTextNode } from "@/util/editorRedactionUtils";
+import { ACCEPTED_PREFIX } from "../util/editorRedactionUtils";
 
+export const toPDF = (body) => {
+    return Array.from(body.document.documentBody)
+        .map((paragraph) => {
+            const p_tag_start = "<p>";
+            const p_tag_end = "</p>";
+            const internal_text = paragraph.children.map((node) => {
+                const isAccepted = getRedactionsOnTextNode(node, ACCEPTED_PREFIX).size > 0;
+                let text = node.text.replace("\n", p_tag_end + "<br>" + p_tag_start)
```

Comment:
const text

Reply:
also nit but maybe put the text code inline into the ternary statement to avoid calling it if we don't need to because the replace can be a little expensive

Reply:
and also call `.trim()` or whatever bc there's some weird stuff happening w spaces rn

Reply:
trim() doesn't fix the space issues, not sure what's causing it (will look into)